tools/flaky-test-reporter/result.go

```diff
@@ -165,3 +166,23 @@ func collectTestResultsForRepo(jc *JobConfig) (*RepoData, error) {
 	}
 	return rd, nil
 }
+
+// getLatestFinishedBuilds is an inexpensive way of listing latest finished builds, in comparing to
+// the GetLatestBuilds function from prow package, as it doesn't precompute start/finish time before sorting.
+// This function takes the assumption that build IDs are always incremental integers
+func getLatestFinishedBuilds(job *prow.Job, count int) []prow.Build {
+	var builds []prow.Build
+	buildIDs := job.GetBuildIDs()
+	fmt.Println(len(buildIDs))
+	sort.Sort(sort.Reverse(sort.IntSlice(buildIDs)))
+	for _, buildID := range buildIDs {
+		if len(builds) >= count {
+			break
+		}
+		build := job.NewBuild(buildID)
+		if nil != build.FinishTime {
+			builds = append(builds, *build)
+		}
+	}
+	return builds
```

Comment:
You could be paranoid here (since you're relying on the assumption that IDs are increasing integers) and ensure that the start time of the builds is indeed decreasing. Since you already computed the start/end time, and you're gonna iterate over `count` items, this won't make your function slower. It would be a clear signal if someday things change radically in the GCS bucket.

Reply:
Good idea, implemented guard against it